File: frontend/src/lib/billing.ts
import { api } from "@/lib/api";

export type BillingStatus = {
  current_plan: string;
  subscription_status: string;
  can_create_more_deals: boolean;
  max_deals: number | null;
  deals_used: number;
  is_pro: boolean;
};

export async function fetchBillingStatus(): Promise<BillingStatus> {
  return api.getBillingStatus();
}

File: backend/app/schemas/billing.py
from pydantic import BaseModel


class BillingStatusResponse(BaseModel):
    current_plan: str
    subscription_status: str
    can_create_more_deals: bool
    max_deals: int | None
    deals_used: int
    is_pro: bool


class CheckoutSessionResponse(BaseModel):
    checkout_url: str


class BillingPortalResponse(BaseModel):
    portal_url: str


class BillingConfirmResponse(BaseModel):
    current_plan: str
    subscription_status: str
    is_pro: bool